src/main/java/core/basesyntax/strategy/operation/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void performOperation(FruitTransaction transaction) {
+        int currentQuantity = Storage.storage.get(transaction.getFruitName());

```

Comment:
There's a potential NullPointerException here. If `Storage.storage.get(transaction.getFruitName())` returns `null` because the fruit name doesn't exist in the storage yet, this line will throw an exception. You should check if the fruit exists in the storage and handle the case where it doesn't by initializing the quantity to zero.